src/Services/CleanupService.php

```diff
@@ -0,0 +1,390 @@
+{
+
+    private $path;
+
+    private $em;
+    private $logger;
+
+    private $inactiveDevices;
+    private $activeDeviceIds;
+    private $storedDeviceIds;
+    private $storedActiveDevices;
+
+    private $storageNode;
+    private $activeProbes;
+    private $activeGroups;
+
+
+    public function __construct(EntityManagerInterface $em, LoggerInterface $logger, ParameterBagInterface $params)
+    {
+        $this->path = $params->get('rrd_storage_path');
+
+        $this->em = $em;
+        $this->logger = $logger;
+
+    }
+
+    public function setStorageNode(StorageNode $storageNode){
+        $this->storageNode = $storageNode;
+    }
+
+    public function cleanup(){
+
+        echo 'Sitrep ' . PHP_EOL;
+
+        $this->logger->info('Retrieving statistics...');
+        $this->getCurrentSituation();
+
+        echo 'Remove Inactive Devices ' . PHP_EOL;
+
+        $this->logger->info('Removing inactive devices...');
+        $this->removeInactiveDevices(10);
+
+        echo 'Sitrep ' . PHP_EOL;
+
+        $this->logger->info('Updating statistics...');
+        $this->getCurrentSituation();
+
+        echo 'Gather Active Probes ' . PHP_EOL;
+
+        $this->logger->info('Gathering active probes...');
+        $this->gatherActiveProbes();
+
+        echo 'Remove Inactive Probes ' . PHP_EOL;
+
+        $this->logger->info('Removing inactive probes...');
+        $this->removeInactiveProbes();
+
+        echo 'Gather Active Slave groups ' . PHP_EOL;
+
+        $this->logger->info('Gathering active slave groups...');
+        $this->gatherActiveSlaveGroups();
+
+        echo 'Remove Inactive Slave Groups ' . PHP_EOL;
+
+        $this->logger->info('Removing inactive slave groups...');
+        $this->removeInactiveSlaveGroups();
+    }
+
+    protected function getCurrentSituation()
+    {
+
+        //create an array of existing directories
+        $this->storedDeviceIds = $this->getDirContent($this->path);
+
+        //remove newline
+        array_pop($this->storedDeviceIds);
+
+        //get only the active devices based on previous result set
+        $this->storedActiveDevices = $this->em->getRepository(Device::class)->findBy(['id' => $this->storedDeviceIds]);
+
+        //get an array of id's for reference from the devices that are active
+        //?? is this better than looping over each one of the storedActiveDevices ??
+        $activeDevices = $this->em->createQuery('SELECT d.id FROM App:Device d')->getArrayResult();
+
+        //Do the comparison to get the inactive devices
+        $this->activeDeviceIds = array_column($activeDevices, 'id');
+        $this->inactiveDevices = array_diff($this->storedDeviceIds, $this->activeDeviceIds);
+
+        echo 'Stored Devices: ' . $this->getStoredDeviceCount() . PHP_EOL;
+        echo 'Active Devices: ' . $this->getActiveDeviceCount() . PHP_EOL;
+        echo 'Inactive Devices: ' . $this->getInactiveDeviceCount() . PHP_EOL;
+    }
+
+
+    protected function gatherActiveSlaveGroups(){
```

Comment:
Can this be private?
Indicate your return type.
Use consistent naming: `get` and `gather` are conceptually the same, and you've already used `get` previously.

This seems like a `set` action?